python3-sys/src/initconfig.rs

```diff
@@ -14,10 +14,16 @@ pub enum PyStatusType {
 #[repr(C)]
 #[derive(Copy, Clone)]
 pub struct PyStatus {
-    _type: PyStatusType,
-    func: *const c_char,
-    err_msg: *const c_char,
-    exitcode: c_int,
+    pub _type: PyStatusType,
+    pub func: *const c_char,
+    pub err_msg: *const c_char,
+    pub exitcode: c_int,
+}
+
+impl Default for PyStatus {
+    fn default() -> Self {
+        unsafe { core::mem::zeroed() }
```

Comment:
Since `PyStatus` is an `enum`, and enums are prime examples of UB in the [doc behind zeroed()](https://doc.rust-lang.org/nightly/core/mem/union.MaybeUninit.html#method.zeroed), I think this warrants at least a comment. I think in this case it's ok, because 0 is indeed among the target values in the enum definition.

Also maybe, mention that `NULL` is an admissible value for these pointers on the Python side (I understand we're in FFI unsafe territory anyway).

Reply:
I changed the code here to avoid UB.

As for whether `Default` is needed, the typical pattern here is:

```
let mut pre_config = pyffi::PyPreConfig::default();
unsafe {
    pyffi::PyPreConfig_InitIsolatedConfig(&mut pre_config);
}
```

You need to obtain a struct instance to pass to a Python API which will "initialize" it. It doesn't really matter what the initial values are, since the Python APIs populate all fields AFAICT.

The Python APIs simply set default values for a given profile. It will be common for callers to want to update those fields before the struct is used to initialize the interpreter.

Reply:
The way I read the docs for `zeroed()`, it's not UB if 0 is a valid value (which it is, because the enum has a `0` value and it's `repr(C)` for good measure).

That said, I don't think there's any harm to adding the extra set.  The compiler ought to elide it anyway.

Reply:
@markbt, that's what I thought, too, yes.